tests/xharness/Xharness.Tests/Execution/Tests/ProcessManagerTests.cs

```diff
@@ -0,0 +1,178 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using NUnit.Framework;
+using Xharness.Execution;
+using Xharness.Logging;
+
+namespace Xharness.Tests.Execution.Tests {
+
+	[TestFixture]
+	public class ProcessManagerTests {
+
+		// not very portable, but we want to make sure that the class does the right thing
+		// maybe in the futue we could create a dummy process that will help us
+		string logPath;
+		string stdoutLogPath;
+		string stderrLogPath;
+		string stdoutMessage;
+		string stderrMessage;
+		Mock<ILogs> logs;
+		ILog executionLog;
+		ILog stdoutLog;
+		ILog stderrLog;
+		string dummyProcess;
+		Process testProcess;
+		ProcessManager manager;
+		Dictionary<string, string> environmentVariables = new Dictionary<string, string> ();
+
+		[SetUp]
+		public void SetUp ()
+		{
+			logPath = Path.GetTempFileName ();
+			stderrLogPath = Path.GetTempFileName ();
+			stdoutLogPath = Path.GetTempFileName ();
+			stdoutMessage = "Hola mundo!!!";
+			stderrMessage = "Adios mundo cruel";
+			logs = new Mock<ILogs> ();
+			executionLog = new LogFile (logs.Object, "my execution log", logPath);
+			stdoutLog = new LogFile (logs.Object, "my stdout log", stdoutLogPath);
+			stderrLog = new LogFile (logs.Object, "my stderr log", stderrLogPath);
+			dummyProcess = Path.Combine (Path.GetDirectoryName (GetType ().Assembly.Location), "DummyTestProcess.exe");
+			manager = new ProcessManager ();
+			testProcess = new Process ();
+			testProcess.StartInfo.FileName = "mono";
+		}
+
+		[TearDown]
+		public void TearDown ()
+		{
+			executionLog?.Dispose ();
+			executionLog = null;
```

Comment:
I think NUnit creates new instance of the test class for each test so nullying everything might not be needed. But maybe I am wrong and that's XUnit or VSTest

Reply:
It will not create a new instance of the class, but will call SetUp instead. xUnit is the one that works like that (and that is why it does not have a SetUp but uses constructors)